### PR TITLE
Server endpoint basic responsiveness tests

### PR DESCRIPTION
**Pull request by brianprice88**
Moved around some files too.  Tried to test the db by inserting mock data then running GET requests through server, but kept 'not recognized' errors w/ regard to my async syntax (e.g. before), so I'm not sure what I need to download for those to run.

### CODE REVIEW

**Review by akuo34 — COMMENTED** [review]
Were you able to get your test working? i saw that you moved your app.listen to a separate file. I also did that because I read somewhere that that's how you fix the "port already in use" issue. But then because I did that. I could no longer access my FE in the browser. I found another workaround if that's an issue for you.

**Comment by brianprice88** [comment]
Yeah i got those tests working just checking status response and like if you write a generic string response for a certain endpoint, however I couldn't get it to work when actually trying my database functions.  One error was that you couldn't have console.logs (e.g. in database connection), and also the 'before' syntax I guess only works with Mocha (not Jest), but even when I took those out, I couldn't get the test to work.

**Comment by brianprice88** [comment]
but yeah moving files around helped a bit - i referenced the old review/sprints and the structure of each file (to prevent server/database connection interacting with the tests)...but that still didn't get me very far, sadly.

**Comment by akuo34** [comment]
Yea I had the same console.log errors so I commented out the db connection console logs
if you have trouble connecting to your FE after moving the app.listen out. I found this online which worked for me: 
in your server/index.js file: add an if statement instead of moving it to another file

if (process.env.NODE_ENV !== 'test') {
  app.listen(port, () => console.log(`listening on port ${port}`));
}

and then in your package.json: add this as the test script

"test": "NODE_ENV=test jest --forceExit"

**Comment by akuo34** [comment]
it works with Postman though? just not through the test script?

File: test/server.test.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+// //     server.close();
+//   });
+// });
+
+describe('server api', () => {
+  it('gets all the trips in the database', (done) => {
+    request
+      .get('/api/trips')
+      .expect(200);
+    done();
```
Comment (akuo34) [inline test/server.test.js:65]: is done() being called before your promise chain ends? I'm using async await with my tests.

**Comment by brianprice88** [comment]
> Yea I had the same console.log errors so I commented out the db connection console logs
> if you have trouble connecting to your FE after moving the app.listen out. I found this online which worked for me:
> in your server/index.js file: add an if statement instead of moving it to another file
> 
> if (process.env.NODE_ENV !== 'test') {
> app.listen(port, () => console.log(`listening on port ${port}`));
> }
> 
> and then in your package.json: add this as the test script
> 
> "test": "NODE_ENV=test jest --forceExit"

thanks i'll look into that when i go back to testing - haven't really tried much front end testing aside from basic React/Enzyme e.g. checking components exit (though even that wasn't working), so I don't think I'm up to client/server/db connections.  Re: Postman yes that works fine.

File: test/server.test.js (hunk 1 above)
Comment (brianprice88) [reply test/server.test.js:65]: The 'expect(200)' passed so I didn't think so - also I should add that when I was trying tests for content-length and the like, those were passing through but it was getting binary data even when I tried to parse it.  I did use async/await at one point and watched some tutorials using it, but still that wasn't working for me.

**Pull request merged**